addons/web/static/src/js/view_list.js

```diff
@@ -1872,7 +1872,7 @@ var Record = instance.web.Class.extend(/** @lends Record# */{
             if (typeof val !== 'object') {
                 output[k] = val;
             } else if (val instanceof Array) {
-                output[k] = val[0];
+                output[k] = val.length > 0 ? val[0] : null;
```

Comment:
OK, let's see what the Odoo's expert says (maybe he/she has another suggestion).